### DISCUSSION TITLE
Qucs-S 1.0.0

### DISCUSSION THREAD

**Discussion by ra3xdh** [ra3xdh/qucs_s#175] [Announcements]
## About this release

I am glad to present Qucs-S v1.0.0! Switching to v1.0.0 version number indicates that Qucs-S is a mature software ready for professional usage. Thanks Tom Hajjar @tomhajjar and Sergey Krasilnikov  @Zergud for testing and contribution! 

## New features

* Added basic support for logic gates simulation (Ngspice >= 38) required #97
* Added possibility to set engineering notation (like 1k, 1M, etc.) for numbers on diagrams #92
* Added symbol preview and search feature for component library in main windows #135
* Backported qucs-attenuator, qucs-transcalc and qucs-powercombining from Qucs-0.0.20 #142
* Added possibility to set dB units for Y-axis on digrams from the diagram properties dialog #92
* Backported advanced features for projects tree from Qucs-0.0.20 #95
* Show warning if schematic file name changed, but dataset/display file name are not updated #111
* Revised examples tree (by Tom Hajjar @tomhajjar); legacy Ngspice-incompatible examples 
  moved to `examples/qucsator` subdirectory

## Component library

* Qucs-lib tool has been removed from the application; all its features are
  availbale now from the main window left panel #139
* Fixed Tubes.lib and LEDs.lib libraries (by @tomhajjar)

## General improvements

* Fixed numerous bug reports. Thanks @tomhajjar and @Zergud for testing and providing patches.

## Windows

* Added 32-bit Windows portable package

## Known issues

* Parameter sweep doesn't work when using the Fourier simulation with XYCE backend #73
* Netlister may produce duplicate .SUBCKT entry when using the same component from user and system library #82
* Auto-scaling on diagram may fail on some conditions for Windows 64-bit platform #133 Other platforms are not affected
* The 1-point AC simulation produce a set of 1x1 plot if ParameterSweep instead of smooth curve is attached #156

## Additional information

See also our presentation from MIXDES-2017 conference for more info about Qucs-S and its compact modelling features. You may download the [Part 1](https://www.researchgate.net/publication/318099550_An_outline_of_Qucs-S_compact_device_modelling_History_and_capabilities_Part_1_Equation-Defined_Device_EDD_modelling_to_Verilog-A_module_synthesis) and [Part2](https://www.researchgate.net/publication/318099550_An_outline_of_Qucs-S_compact_device_modelling_History_and_capabilities_Part_1_Equation-Defined_Device_EDD_modelling_to_Verilog-A_module_synthesis) from the ResearchGate.

### Source code

Download the source code file `qucs-s-1.0.0.tar.gz` and build it with `cmake` to try new Qucs-S. 

### Linux

* AppImage for all Linux distributions could be found here: https://download.opensuse.org/repositories/home:/ra3xdh/AppImage/
*  DEB and RPM packages for Debian, Ubunut, Fedora, CentOS, openSUSE generated with OBS. Find the repository for your distribution here: https://software.opensuse.org/download.html?project=home%3Ara3xdh&package=qucs-s

### Windows

Windows portable package could be downloaded as zipped distribution. Unpack and launch `qucs-s.exe` from `bin` subdirectory. Ngspice requires a separate installation. There is two windows packages available:
* `qucs_s_win64_1_0_0.zip` 64-bit (x86_64) portable Windows version;
* `qucs_s_win32_1_0_0.zip` 32-bit (i686) portable Windows version;

<hr /><em>This discussion was created from the release <a href='https://github.com/ra3xdh/qucs_s/releases/tag/1.0.0'>Qucs-S 1.0.0</a>.</em>

**Comment by tomhajjar** [comment]
The xyce examples in the following folder still have errors.
https://github.com/ra3xdh/qucs_s/tree/master/examples/xyce/Xyce_Examples

I uploaded replacements that I updated to work and improved the graphics.
Manual scaling was also tuned on to work in Windows. xyce.zip
<https://github.com/ra3xdh/qucs_s/files/9627723/xyce.zip>
https://github.com/ra3xdh/qucs_s/issues/162

I don't want to step on the author Tom Russo's toes. He has updated the
files on his github page.
https://github.com/tvrusso/Xyce_qucs-s_Examples

**Comment by ra3xdh** [comment]
> The xyce examples in the following folder still have errors. https://github.com/ra3xdh/qucs_s/tree/master/examples/xyce/Xyce_Examples

The examples from this path are taken from the Xyce regression suite. I have added them to the examples tree, but the better solution may be not to include these examples in the Qucs-S package. It may be better to provide somewhere a link to Xyce Regression repository instead.

**Comment by ra3xdh** [comment]
> Mr Russo does not seem interested in maintaining the files. 

The latest commit is this repository https://github.com/tvrusso/Xyce_qucs-s_Examples/ was in August 2022, so it seems to be maintained. Duplication of the Xyce regression suite in the Qucs-S examples may be not a good idea. See also the log of changes: https://github.com/tvrusso/Xyce_qucs-s_Examples/commits/master

**Reply by tvrusso** [reply]
Xyce_qucs-s_Examples is not the Xyce regression suite, it's just a handful of toys I created to help Xyce users use Qucs-s.  There are only 11 examples in there.  I try to keep them working as Qucs-s and Xyce development goes on, but mostly they don't need much maintenance.  I am not developing any more examples in that repo.

The Xyce regression suite has no Qucs-s schematics, but has thousands of test cases to keep Xyce development from breaking important use cases.  Some of those test cases are useful examples, but that's not its purpose.

**Comment by SilvanRehm** [comment]
> Source code
Download the source code file **qucs-s-0.0.24.tar.gz** and build it with cmake to try new Qucs-S.

shouldn't this be qucs-s-1.0.0.tar.gz ?

**Comment by ra3xdh** [comment]
> shouldn't this be qucs-s-1.0.0.tar.gz ?

Yes, I have just fixed this typo. It should be 1.0.0
